scheduler/pkg/envoy/processor/incremental.go

```diff
@@ -588,6 +588,7 @@ func (p *IncrementalProcessor) modelUpdate(modelName string) error {
 			logger.Debugf("sync: No server - removing for %s", modelName)
 			if err := p.removeRouteForServerInEnvoyCache(modelName); err != nil {
 				logger.WithError(err).Errorf("Failed to remove model route from envoy %s", modelName)
+				p.modelStore.UnlockModel(modelName)
```

Comment:
nice catch!

Reply:
we might be able to convert this into a defer, as we always unlock before return

Reply:
there is a for loop that goes over multiple models and therefore we want to lock/unlock per iteration. maybe we should refactor this logic per iteration in a separate function so that we can use defer.

Reply:
It looks like we need to release the model lock before running further logic at the end of this method